contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -281,10 +298,51 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoTriggerCount(uint256 newAutoTriggerCount) external onlyAdmin {
+        uint256 prevAutoTriggerCount = _autoTriggerCount;
+        if (prevAutoTriggerCount == newAutoTriggerCount) {
+            return;
+        }
+
+        _autoTriggerCount = newAutoTriggerCount;
+
+        emit AutoTriggerCountUpdated({
+            prevAutoTriggerCount: prevAutoTriggerCount,
+            newAutoTriggerCount: newAutoTriggerCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function trigger() external nonReentrant {
+        address[] memory values = _pools.values();
+        uint256 numOfPools = values.length;
+        uint256 startIndex = _nextTriggerIndex;
+        uint256 count = Math.min(numOfPools, _autoTriggerCount);
+
+        for (uint256 i = 0; i < count; i++) {
+            uint256 index = (startIndex + i) % numOfPools;
+            _processRewards(Token(values[index]), true);
```

Comment:
Perhaps `_processRewards` should return whether the program was processed or not (== skipped) and then it shouldn't be taken into account and the next program should be considered instead?

Reply:
good idea + will make the cost/reward more consistent

Reply:
done